masq/src/test_utils/mocks.rs

```diff
@@ -352,9 +393,166 @@ impl TestStreamFactoryHandle {
                     }
                     thread::sleep(Duration::from_millis(100));
                 }
-                Err(_) => break,
+                Err(_) => {
+                    break;
+                }
             }
         }
         accum
     }
 }
+
+#[derive(Clone)]
+pub struct MixingStdout {
+    channel_half: Sender<String>,
+}
+
+impl MixingStdout {
+    pub fn new(sender: Sender<String>) -> Self {
+        MixingStdout {
+            channel_half: sender,
+        }
+    }
+}
+
+impl Write for MixingStdout {
+    fn write(&mut self, buf: &[u8]) -> std::io::Result<usize> {
+        self.channel_half
+            .send(std::str::from_utf8(buf).unwrap().to_string())
+            .unwrap();
+        Ok(0)
+    }
+
+    fn flush(&mut self) -> std::io::Result<()> {
+        Ok(())
+    }
+    fn write_fmt(&mut self, fmt: Arguments<'_>) -> std::io::Result<()> {
+        self.channel_half.send(fmt.to_string()).unwrap();
```

Comment:
Maybe this should call self.write()?

Reply:
That may be more complicated than now in the end.